### ISSUE TITLE
Performance issue with `Timer`

### ISSUE THREAD

**Issue by Byron** [stjepang/async-io#6]
This 'regression' appears a while ago in `criner`, and apparently went unnoticed when switching from `tokio::time::Delay` to `async_io::Timer`.

<img width="1432" alt="Screenshot 2020-07-24 at 10 17 31" src="https://user-images.githubusercontent.com/63622/88357075-9795f180-cd9c-11ea-8f9e-39d1e63a47c4.png">

The picture above shows a `criner` process doing nothing, consuming 10 to 15% of CPU for displaying 4 progress bars which update once per second. The expensive part of this used to be the drawing itself, but now it appears to be dominated by the condvar in `parking`.

In `smol` 0.1.5 using `smol::Timer`, this looked like that:

<img width="1440" alt="Screenshot 2020-07-24 at 11 00 55" src="https://user-images.githubusercontent.com/63622/88357569-5a326380-cd9e-11ea-9a19-f99a46cbe77d.png">

All that shows up is the drawing of the TUI.

Having had a look myself I am absolutely puzzled as to why that is - the code looks very similar.

**Any hint at how to solve this would be greatly appreciated.**
Thank you

<details>

#### Meta

* ['utils.rs' on a criner branch that works as expected, without performance issues](https://github.com/the-lean-crate/criner/blob/378f7d28bf8591ecaca82cd1f74c382eb9929b9d/criner/src/utils.rs#L41)
   * It's noteworthy that `criner` didn't change in the relevant portions of the code [responsible for the timer invocations](https://github.com/the-lean-crate/criner/compare/without-performance-regression...master#diff-1dcf5476e374956c4bac4ecfde88ea32) _see utils.rs_.

<img width="898" alt="Screenshot 2020-07-24 at 11 20 58" src="https://user-images.githubusercontent.com/63622/88357989-daa59400-cd9f-11ea-9254-354723f64e94.png">

* [the version of 'utils.rs' that suffers from 'slow' timers](https://github.com/the-lean-crate/criner/blob/7c27b064c1d3359daff8ef3cf70c72474211be70/criner/src/utils.rs#L43)

**Comment by stjepang** [comment]
This should now be fixed in v0.1.7 - can you check?

**Comment by Byron** [comment]
Thanks a lot for your help! I think fixing this one will be important so that async-std doesn't get swamped with similar performance related tickets _if_ they use the same timer implementation.

v0.1.7 It's a 'little' better, in that the CPU now sometimes drops under 10% to 8%, whereas usually it would be quite consistently at 11 to 12%.

A 12 second sample still yields a similar image as before (and for the uninitiated: TUI draw time should dominate the CPU, not the few 4 to 5 timers the application uses):

<img width="1437" alt="Screenshot 2020-07-29 at 08 17 14" src="https://user-images.githubusercontent.com/63622/88742200-355e3780-d174-11ea-872b-9d9c52b032b8.png">

**Comment by Byron** [comment]
#### How to easily reproduce this

```bash
git clone https://github.com/crates-io/prodash
cd prodash
cargo run --example dashboard --release  --features="tui-renderer tui-renderer-crossterm line-renderer line-renderer-crossterm ctrlc" 
```

The above program when profiled looks very much liker `criner`, in that the async-io code dominates the runtime. This one uses way more timers, so it's probably a great example to get fast (again).

<img width="1440" alt="Screenshot 2020-07-29 at 08 45 30" src="https://user-images.githubusercontent.com/63622/88743776-1cf01c00-d178-11ea-9618-931cc48ebcf2.png">

**Comment by stjepang** [comment]
Fixed in v0.1.8
